docs/docs/recipes/integrate-logto/go-web.mdx

```diff
@@ -0,0 +1,352 @@
+import ConfigurePostSignOutRedirectUri from './fragments/_configure-post-sign-out-redirect-uri.mdx';
+import ConfigureRedirectUri from './fragments/_configure-redirect-uri.mdx';
+import FurtherReadings from './fragments/_further-readings.md';
+import GetAppSecret from './fragments/_get-app-secret.mdx';
+import SignInFlowSummary from './fragments/_web-sign-in-flow-summary.mdx';
+
+# Go Web: Integrate Logto
+
+<AppNote type="Traditional web" />
+
+## Add Logto SDK as a dependency
+
+:::tip
+We demonstrate the integration process of the SDK based on the <a href="https://gin-gonic.com">Gin Web Framework</a>. You can also easily integrate Logto into other web frameworks by taking the same steps.
+In the following code snippets, we assume your app is running on <code>http://localhost:8080</code>.
+:::
+
+Assuming your application code looks like as below:
+
+```go
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+func main() {
+	router := gin.Default()
+	router.GET("/", func(ctx *gin.Context) {
+		ctx.String(200, "Hello Logto!")
+	})
+	router.Run(":8080")
+}
+```
+
+Execute in the project directory:
+
+```bash
+go get github.com/logto-io/go
+```
+
+Add the `github.com/logto-io/go/client` package to your application code:
+
+```go
+// main.go
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+	// Add dependency
+	"github.com/logto-io/go/client"
+)
+
+func main() {
+	router := gin.Default()
+	router.GET("/", func(c *gin.Context) {
+		c.String(200, "Hello Logto!")
+	})
+	router.Run(":8080")
+}
+```
+
+## Create storage for Logto to store user credentials
+
+Logto stores user credentials in the `Storage`.  
+For traditional web applications, we need to store the user credentials in the session of the request made by the user.  
+Although Go itself does not provide an implementation of `session`, we can use the Logto SDK's `Storage` type to create a session-based `Storage` adapter.
+
+The `Storage` type in the Logto SDK is as follows:
+
+```go
+// github.com/logto-io/client/storage.go
+package client
+
+type Storage interface {
+	GetItem(key string) string
+	SetItem(key, value string)
+}
+```
+
+We will to implement our Storage based on [github.com/gin-contrib/sessions](https://github.com/gin-contrib/sessions).
+
+Create a `session_storage.go` file, define a `SessionStorage` and implement the Logto SDK's `Storage` interfaces:
+
+```go
+// session_storage.go
+package main
+
+import (
+	"github.com/gin-contrib/sessions"
+)
+
+type SessionStorage struct {
+	session sessions.Session
+}
+
+func (storage *SessionStorage) GetItem(key string) string {
+	value := storage.session.Get(key)
+	if value == nil {
+		return ""
+	}
+	return value.(string)
+}
+
+func (storage *SessionStorage) SetItem(key, value string) {
+	storage.session.Set(key, value)
+	storage.session.Save()
+}
+```
+
+## Prepare required middlewares
+
+Next, use the [github.com/gin-contrib/sessions](https://github.com/gin-contrib/sessions) middleware in the application:
```

Comment:
Should we list what kinds of middlewares are needed here?

P.S.  `gin-contrib/sessions` is referenced here again as a middleware. What is the relationship compared to the session storage block above? Should this also part of the session storage provider?